Aiolos/Aiolos/Sources/PanelGestures.swift

```diff
@@ -73,302 +85,452 @@ extension PanelGestures: UIGestureRecognizerDelegate {
-            if self.gestureRecognizer(pan, isWithinContentAreaOf: contentViewController), let scrollView = self.verticallyScrollableView(of: contentViewController, interactingWith: pan) {
-                pan.startMode = .onVerticallyScrollableArea(competingScrollView: scrollView)
+        
+        @objc
+        func handlePan(_ pan: UIPanGestureRecognizer) {
+            switch pan.state {
+            case .began:
+                self.handlePanStarted(pan)
+            case .changed:
+                self.handlePanChanged(pan)
+            case .ended:
+                self.handlePanEnded(pan)
+            case .cancelled:
+                self.handlePanCancelled(pan)
+            default:
+                break
+            }
+        }
+        
+        private func handlePanStarted(_ pan: UIPanGestureRecognizer) {
+            self.gestures.updateResizeHandle()
+        }
+        
+        private func handlePanChanged(_ pan: UIPanGestureRecognizer) {
+            guard let parentView = self.panel.parent?.view else { return }
+            
+            func dragOffset(for translation: CGPoint, moveAllowed: Bool) -> CGFloat {
+                let dragCoefficient: CGFloat = 1/5 // TODO: revise
+                return moveAllowed ? translation.x : translation.x * dragCoefficient
+            }
+            
+            let translation = pan.translation(in: parentView)
+            let transformation = CGAffineTransform(translationX: translation.x, y: 0)
+            let targetFrame = self.panel.view.frame.applying(transformation)
+            let moveAllowed = self.panel.animator.notifyDelegateAboutMove(to: targetFrame)
+            let xOffset = dragOffset(for: translation, moveAllowed: moveAllowed)
+            guard xOffset != 0.0 else { return }
+            
+            self.panel.animator.performWithoutAnimation { self.panel.view.transform = CGAffineTransform(translationX: xOffset, y: 0) }
+        }
+        
+        private func handlePanEnded(_ pan: UIPanGestureRecognizer) {
+            guard let parentView = self.panel.parent?.view else { return }
+            
+            // The view is being transformed, thus the frame changes while dragging, hence the correction
+            let originalFrame = self.panel.view.frame.applying(self.panel.view.transform.inverted())
+            let offset = pan.translation(in: parentView).x
+            let velocity = pan.velocity(in: parentView).x
+            let context = PanelTransitionCoordinator.HorizontalTransitionContext(panel: self.panel, parentView: parentView, originalFrame: originalFrame, offset: offset, velocity: velocity)
+            self.panel.animator.notifyDelegateOfMove(from: originalFrame, to: self.panel.view.frame, context: context)
+
+            let initialVelocity = self.initialVelocity(with: context)
+            let timing = Animation.overdamped.makeTiming(with: initialVelocity)
+            self.panel.animator.animateWithTiming(timing, animations: {
```

Comment:
nit: not sure if it works because of the completion block, but maybe we can use trailing closure syntax here?
```suggestion
            self.panel.animator.animateWithTiming(timing) {
```